services/rfq/relayer/inventory/rebalance.go

```diff
@@ -252,3 +254,30 @@ func getRebalanceAmount(span trace.Span, cfg relconfig.Config, tokens map[int]ma
 
 	return amount, nil
 }
+
+// getTotalBalance calculates the total balance for a token
+// across all chains that support the given rebalance method.
+func getTotalBalance(cfg relconfig.Config, tokens map[int]map[common.Address]*TokenMetadata, tokenName string, method relconfig.RebalanceMethod) *big.Int {
+	totalBalance := big.NewInt(0)
+	for _, tokenMap := range tokens {
+		for _, tokenData := range tokenMap {
+			if tokenData.Name != tokenName {
+				continue
+			}
+			if supportsRebalanceMethod(cfg, tokenData.ChainID, tokenData.Addr.Hex(), method) {
+				totalBalance.Add(totalBalance, tokenData.Balance)
+			}
+		}
+	}
+	return totalBalance
+}
+
+func supportsRebalanceMethod(cfg relconfig.Config, chainID int, addr string, method relconfig.RebalanceMethod) bool {
```

Comment:
This feels generic enough to be located somewhere else, but I'm not entirely sure. Would trust the Go masters judgement here. In any case, could use a few words in the docs.

Reply:
Yeah, I mostly split this out to avoid dedup and make other functions a bit less verbose, I think it's probably fine to leave in this file as long as it's not used elsewhere